src/main/java/carpet/mixins/ServerPlayNetworkHandler_scarpetEventsMixin.java

```diff
@@ -240,8 +240,8 @@ private void onEntityInteract(PlayerInteractEntityC2SPacket playerInteractEntity
     ))
     private void onEntityAttack(PlayerInteractEntityC2SPacket playerInteractEntityC2SPacket_1, CallbackInfo ci)
     {
-        //todo add hit and hand in the future
-        PLAYER_ATTACKS_ENTITY.onEntityHandAction(player, playerInteractEntityC2SPacket_1.getEntity(player.getServerWorld()), null);
+        //todo add hit in the future
+        PLAYER_ATTACKS_ENTITY.onEntityHandAction(player, playerInteractEntityC2SPacket_1.getEntity(player.getServerWorld()), playerInteractEntityC2SPacket_1.getHand());
```

Comment:
remove that - this all changed in 21w08a. No need to revise that now.

Again, too many small changes for one PR

Reply:
sorry.